curriculum/challenges/english/25-front-end-development/workshop-binary-search/684165a72e33e26a40904f8b.md

```diff
@@ -0,0 +1,55 @@
+---
+id: 684165a72e33e26a40904f8b
+title: Step 13
+challengeType: 20
+dashedName: step-13
+---
+
+# --description--
+
+Finally, if the `value` is not found at the middle and is also not greater than `value_at_middle`, then the `value` must be less than `value_at_middle`, and must be on the left. The `else` block will handle this.
+
+So, finish up the loop by removing the `break` keyword and adding an `else` block. Inside it, update the `high` variable by subtracting `1` from `mid`.
+
+# --hints--
+
+You should replace the `break` keyword with an else block that updates the `high` variable to `mid - 1`.
+
+```js
+({
+    test: () => 
+        assert(
+            runPython(`
+                _Node(_code).find_function("binary_search").find_whiles()[0].find_ifs()[0].is_equivalent("if value == value_at_middle:\\n    return path_to_target\\nelif value > value_at_middle:\\n    low = mid + 1\\nelse:    high = mid - 1")`
```

Comment:
Same case here, we should reach the `else` clause with `_Node(_code).find_function("binary_search").find_whiles()[0].find_ifs()[0].find_bodies()[2]`

If you want to be sure that it's an `else` instead of an `elif` you should check how many elements are in `_Node(_code).find_function("binary_search").find_whiles()[0].find_ifs()[0].find_conditions()`. With one `if`, one `elif` and one `else` --> two conditions